### PR TITLE
fix: validate username on create_user (#276)

### LINKED ISSUE #276: [Tech Debt] High - create_user accepts usernames of unbounded length with no character validation
`db/src/auth/users.rs:13–14` calls `validate_password(password)?` at the top of `create_user` but has no equivalent `validate_username` call. Passwords enforce a minimum length of 10 characters and a maximum of 128 characters (plus a common-password reject-list) via `db/src/auth/password.rs`. Usernames have no length floor, no length ceiling, no character restriction, and no normalization. A caller can create a username of zero characters, a username of several megabytes, a username consisting entirely of Unicode control characters or null bytes, or a username of only whitespace — all pass the current guard and are stored verbatim in `users.username TEXT NOT NULL COLLATE NOCASE`.

This is **High** severity. A zero-length username creates a nonsensical user record. An unbounded username consumes DB storage and log space proportional to the input; the only cap is the global 1 MiB body limit on the registration request. Null bytes and control characters can corrupt log output and potentially cause downstream display bugs. The `COLLATE NOCASE` dedup at `users.rs:45` guards against exact case-collisions but does not normalize Unicode homoglyphs (e.g. `аdmin` with Cyrillic `а` is distinct from `admin`), so a homoglyph username could be used to impersonate an admin in displayed contexts.

Sources: `db/src/auth/users.rs:13–14` — `create_user` with `validate_password` call but no username validation. `shared/src/lib.rs:548–557` — `RegisterRequest` struct with `username: String` and no validation attributes. `db/src/auth/password.rs` — existing `validate_password` pattern to follow.

Add a `validate_username(username: &str) -> Result<(), AuthError>` function in `db/src/auth/password.rs` (or a new `db/src/auth/username.rs`) that enforces: non-empty, maximum 64 characters, no leading/trailing whitespace, no ASCII control characters (≤ 0x1F or 0x7F), and no null bytes. Call it at the top of `create_user`, matching the `validate_password` pattern. Add unit tests covering the boundary conditions, following the existing test suite in `password.rs`.

### PR DESCRIPTION
**Pull request by seamus-sloan**
## Summary

- `create_user` previously called `validate_password` but had no equivalent guard for usernames — any string (including empty, megabyte-sized, leading/trailing whitespace, or ASCII control characters like NUL/`\t`/`\n`/`\x1f`/`\x7f`) was accepted and stored verbatim in `users.username TEXT NOT NULL COLLATE NOCASE`.
- Adds `validate_username` in `db/src/auth/password.rs` (next to `validate_password` for proximity) enforcing: non-empty, max 64 Unicode scalar values, no leading/trailing whitespace, no ASCII controls (`U+0000..=U+001F`, `U+007F`). Called at the top of `create_user`, matching the `validate_password` pattern.
- Extends `AuthError` with four new variants (`UsernameEmpty`, `UsernameTooLong`, `UsernameWhitespace`, `UsernameInvalidChar`) and maps each to a 400 in `server/src/auth/handlers.rs::auth_error_to_response` so registration surfaces the failure cleanly. The handler match was already exhaustive, so the variants are covered explicitly rather than via a wildcard.

Closes #276

## Test plan

- [x] `cargo fmt`
- [x] `cargo clippy --all-targets -- -D warnings` clean (default features). `--all-features` surfaces pre-existing `omnibus-frontend` failures on `main` (deprecated `web_sys::BlobPropertyBag::type_`, missing `CurrentUser` under the `mobile` feature) that are unrelated to this change.
- [x] `cargo test -p omnibus-db` — 342 passed (15 new username tests: empty, single char, 64-char boundary, 65-char over, leading/trailing/only-whitespace, embedded `\t` / `\n` / `\0` / `\x1f` / `\x7f`, three happy-path cases, plus a `create_user_rejects_invalid_username` integration test confirming the guard runs before any DB work)
- [x] `cargo test -p omnibus` — 128 passed (handler match still exhaustive, no behavior regression)

## Notes

- Unicode homoglyph confusables (e.g. Cyrillic `а` vs Latin `a` impersonating `admin`) are deliberately out of scope here — that's a normalization concern that belongs alongside whichever admin UI surfaces start rendering user-supplied usernames in security-sensitive contexts. The existing `COLLATE NOCASE` dedup only guards exact case-collisions; a confusables pass would need to land in the same change as a display-side normalization to be coherent. Filed as a follow-up consideration in the `validate_username` doc-comment.
- The whitespace check intentionally rejects all-whitespace input under `UsernameWhitespace` (since `trim() != self`) rather than `UsernameEmpty`. Locked in a test so the rejection variant doesn't silently change if the rule order is reshuffled.

https://claude.ai/code/session_01Jr6c66demnTfNf1XYGTtwX

---
_Generated by [Claude Code](https://claude.ai/code/session_01Jr6c66demnTfNf1XYGTtwX)_

### CODE REVIEW

File: server/src/auth/handlers.rs [hunk 1]
```diff
@@ -77,6 +77,24 @@ fn auth_error_to_response(e: AuthError) -> Response {
...
+        AuthError::UsernameWhitespace => (
+            StatusCode::BAD_REQUEST,
+            "username must not have leading or trailing whitespace",
+        )
+            .into_response(),
+        AuthError::UsernameInvalidChar => (
+            StatusCode::BAD_REQUEST,
+            "username contains an invalid control character",
+        )
+            .into_response(),
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply server/src/auth/handlers.rs:97]: Addressed in ca11145 — `register_invalid_username_returns_400` (handlers.rs:394) loops over the six new `Username*` rejection paths (empty, >64 chars, leading/trailing whitespace, ASCII control char, NUL byte), each freshly setting up `rest_router(AppState::new(pool))` and asserting `400 BAD_REQUEST` via `oneshot`.

---
_Generated by [Claude Code](https://claude.ai/code)_
